### PR TITLE
Migration: Adds user blocks to export/import profile

### PR DESCRIPTION
**Pull request by tclaus**
This PR adds archiving and importing the blocks model.
@Flaburgan @cmrd-senya

### CODE REVIEW

File: lib/archive_importer.rb [hunk 1]
```diff
@@ -37,12 +37,26 @@ def create_user(attr)
...
 
   private
 
   attr_reader :archive_hash
 
+  def import_blocks(attr, user)
+    return unless attr[:blocks]
+
+    attr[:blocks].each do |blocked_id|
+      p = Person.find_or_fetch_by_identifier(blocked_id)
```
Comment (SuperTux88) [inline lib/archive_importer.rb:51]: This raises an error when the person isn't known and can't be fetched, which would abort the whole import. Also, if the blocked account was migrated between the export-creation and the import, you want to block the new user, not the old one. So in the end you probably want a similar logic for blocks as it already exists for contacts here: https://github.com/diaspora/diaspora/blob/develop/lib/archive_validator/contact_validator.rb

File: lib/archive_importer.rb [hunk 2]
```diff
@@ -37,12 +37,26 @@ def create_user(attr)
     )
     self.user = User.build(data)
     user.save!
+    import_blocks(attr, user)
```
Comment (SuperTux88) [inline lib/archive_importer.rb:40]: This should be moved to the `import` method above (line 13) where all the other stuff is imported to. If the user already exists `create_user` isn't called, so blocks wouldn't be imported.

File: app/serializers/export/user_serializer.rb [hunk 3]
```diff
@@ -45,6 +46,10 @@ def relayables
       }
     end
 
+    def blocks
+      object.blocks.map(&:person).map(&:diaspora_handle)
```
Comment (SuperTux88) [inline app/serializers/export/user_serializer.rb:50]: I think `.map(&:person).map(&:diaspora_handle)` should be done in one `map` step, to not need to first create a full list of persons. But you can add `diaspora_handle` to the delegated methods in the block class, so you can just do `.map(&:person_diaspora_handle)`.

**Review by SuperTux88 — CHANGES_REQUESTED** [review]
Can you please split this up in two separate pull requests, one for the archive creation additions (for 0.7.x) and one for the import part (for 0.8.x).

Also I think the import part needs more work, to make it more robust, so it works similar to the contacts import (maybe can even be done to share code to have less duplication?).

File: app/serializers/export/user_serializer.rb (hunk 3 above)
Comment (tclaus) [reply app/serializers/export/user_serializer.rb:50]: Today I learned about delegate methods. Thanks.

File: lib/archive_importer.rb (hunk 1 above)
Comment (tclaus) [reply lib/archive_importer.rb:51]: Do you mean the line: ` person.account_migration.newest_person` to fetch migrated user? 

The 'find_or_fetch_by_identifier' method will not throw a exception any more if person can not be fetched, (PR #8257 will introduce this). Many methods rely on returning nil if person is not found, so I globally made this even for callers. 
See:  https://github.com/diaspora/diaspora/blob/d27eefeb3471844179c86c8dc7c28ad99801f40f/lib/archive_validator/contact_validator.rb#L34

Until 8257 a Person-fetch might never return nil if a person is not found.

File: lib/archive_importer.rb (hunk 2 above)
Comment (tclaus) [reply lib/archive_importer.rb]: I had to use both - a user could set this on create or on update an existing user

**Comment by tclaus** [comment]
Closed, because of splitting import/export in two different branches. As requested.

**Pull request closed without merging**